pkg/instancetype/instancetype_test.go

```diff
@@ -1444,4 +1444,240 @@ var _ = Describe("Instancetype and Preferences", func() {
+						Guest: uint32(2),
+					},
+				},
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						CPU: &instancetypev1alpha2.CPUPreferenceRequirement{
+							Guest: uint32(2),
+						},
+					},
+				},
+				nil,
+			),
+			Entry("by an instance type for Memory",
+				&instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+					Memory: instancetypev1alpha2.MemoryInstancetype{
+						Guest: resource.MustParse("1Gi"),
+					},
+				},
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						Memory: &instancetypev1alpha2.MemoryPreferenceRequirement{
+							Guest: resource.MustParse("1Gi"),
+						},
+					},
+				},
+				nil,
+			),
+			Entry("by a VM for vCPUs using PreferSockets (default)",
+				nil,
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					CPU: &instancetypev1alpha2.CPUPreferences{
+						PreferredCPUTopology: instancetypev1alpha2.PreferSockets,
+					},
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						CPU: &instancetypev1alpha2.CPUPreferenceRequirement{
+							Guest: uint32(2),
+						},
+					},
+				},
+				&v1.VirtualMachineInstanceSpec{
+					Domain: v1.DomainSpec{
+						CPU: &v1.CPU{
+							Sockets: uint32(2),
+						},
+					},
+				},
+			),
+			Entry("by a VM for vCPUs using PreferCores",
+				nil,
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					CPU: &instancetypev1alpha2.CPUPreferences{
+						PreferredCPUTopology: instancetypev1alpha2.PreferCores,
+					},
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						CPU: &instancetypev1alpha2.CPUPreferenceRequirement{
+							Guest: uint32(2),
+						},
+					},
+				},
+				&v1.VirtualMachineInstanceSpec{
+					Domain: v1.DomainSpec{
+						CPU: &v1.CPU{
+							Cores: uint32(2),
+						},
+					},
+				},
+			),
+			Entry("by a VM for vCPUs using PreferThreads",
+				nil,
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					CPU: &instancetypev1alpha2.CPUPreferences{
+						PreferredCPUTopology: instancetypev1alpha2.PreferThreads,
+					},
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						CPU: &instancetypev1alpha2.CPUPreferenceRequirement{
+							Guest: uint32(2),
+						},
+					},
+				},
+				&v1.VirtualMachineInstanceSpec{
+					Domain: v1.DomainSpec{
+						CPU: &v1.CPU{
+							Threads: uint32(2),
+						},
+					},
+				},
+			),
+			Entry("by an VM for Memory",
+				nil,
+				&instancetypev1alpha2.VirtualMachinePreferenceSpec{
+					Requirements: &instancetypev1alpha2.PreferenceRequirements{
+						Memory: &instancetypev1alpha2.MemoryPreferenceRequirement{
+							Guest: resource.MustParse("1Gi"),
+						},
+					},
+				},
+				&v1.VirtualMachineInstanceSpec{
+					Domain: v1.DomainSpec{
+						Memory: &v1.Memory{
+							Guest: resource.NewQuantity(1024*1024*1024, resource.BinarySI),
```

Comment:
Any reason you are not using `resource.MustParse("1Gi")` here?

Reply:
`MustParse` doesn't return a pointer, `NewQuantity` does.